tests/models/demultiplexing/test_run_parameters.py

```diff
@@ -38,30 +47,95 @@ def test_run_parameters_parent_class_fails(
     assert "Parent class instantiated" in str(exc_info)
 
 
-def test_run_parameters_hiseq_x(hiseq_x_single_index_run_parameters_path: Path):
-    """Tests that creating a HiSeq RunParameters with a HiSeqX path file works."""
-    # GIVEN a valid HiSeqX run parameters file path
+@pytest.mark.parametrize(
+    "run_parameters_path, constructor, sequencer",
+    [
+        ("hiseq_x_single_index_run_parameters_path", RunParametersHiSeq, Sequencers.HISEQX),
+        ("hiseq_2500_dual_index_run_parameters_path", RunParametersHiSeq, Sequencers.HISEQGA),
+        ("novaseq_6000_run_parameters_path", RunParametersNovaSeq6000, Sequencers.NOVASEQ),
+        ("novaseq_x_run_parameters_path", RunParametersNovaSeqX, Sequencers.NOVASEQX),
+    ],
+)
+def test_run_parameters_init(
+    run_parameters_path: str,
+    constructor: Type[RunParameters],
```

Comment:
Does this lock the type to be the superclass? or what would be the difference to only write RunParameters here?

Reply:
the input here is not an instance of the RunParameters class or any of the child classes, but the class itself. Therefore is `Type[RunParameters]` and not just `RunParameters`.